### DISCUSSION TITLE
.FBX FModel export support?

### DISCUSSION THREAD

**Discussion by OwzygZawg** [4sval/FModel#457] [Q&A]
I'm curious, would it be possible to include a .fbx file export option? Similar to the existing .psk/.psa options.

**Comment by GhostScissors** [comment]
It exists but I don't thing it will work properly and you have to use a different CUE4Parse repo 

https://github.com/FabianFG/CUE4Parse/tree/autodesk-fbx
https://github.com/FabianFG/CUE4Parse/tree/fbx

**Reply by OwzygZawg** [reply]
How do I apply them in FModel?

**Comment by firatgursoy** [comment]
I don't understand the point of exporting materials as JSON. Exporting everything as a single FBX (including mesh, textures, materials, MI, and animations) would be much more practical.

**Reply by Ka1serM** [reply]
It could be implemented if there was a proper FBX Sdk for .Net, but the json export still is needed because unreal materials have an arbitrary amount of named parameters, not like in fbx where its only color, roughness, normal texture.

**Reply by firatgursoy** [reply]
It may be possible to use executables from the SDK while sacrificing performance. Btw, Is there a plugin that will add this JSON output to the unreal editor by redirecting the textures? Is it possible to add the material in the existing json file to the unreal editor in some way?

**Reply by Ka1serM** [reply]
[JsonAsAsset](https://github.com/JsonAsAsset/JsonAsAsset) uses Fmodel jsons to build assets in unreal, but most assets require editor only data in the json which is only found in games like Fortnite with UEFN installed. You can also write a simple script with EditorUtility in Unreal to do this for Materials. About using executables form the SDK, what do you mean exactly? Data for Vertices, Bones etc needs to be passed to build the FBX 🤔

**Reply by firatgursoy** [reply]
I got a lot of errors while importing my materials on JsonAsAsset. I think it needs some time to be hassle-free. 

About sdk, I thought there were some utility executables in the FBX SDK, but there weren't. :) The inputs required to produce an fbx are very complicated. I think creating the materials manually on the editor will be a faster solution for me for now. Data mining is harder than I thought. :)

**Reply by Tectfy** [reply]
> I got a lot of errors while importing my materials on JsonAsAsset. I think it needs some time to be hassle-free.
> 
> About sdk, I thought there were some utility executables in the FBX SDK, but there weren't. :) The inputs required to produce an fbx are very complicated. I think creating the materials manually on the editor will be a faster solution for me for now. Data mining is harder than I thought. :)

What errors? I'd like to help, what game is this?
